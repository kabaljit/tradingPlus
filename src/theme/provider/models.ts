import { ColorPalette } from '../palette.models';
import { IColorContexts } from '../context.models';

export interface ITheme {
  colorPalette: ColorPalette;
  context: {
    colors: IColorContexts;
  };
}

export interface IThemeProvider {
  theme?: ITheme;
}
