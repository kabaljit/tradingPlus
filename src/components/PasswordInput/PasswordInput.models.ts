import { TextInputProps } from '../TextInput/TextInput.models';

export interface IPasswordInputProps extends TextInputProps {
  onPasswordManagerPress?: () => void;
  passwordManagerExists?: boolean;
  showLabel?: string;
  hideLabel?: string;
  state?: string;
}
