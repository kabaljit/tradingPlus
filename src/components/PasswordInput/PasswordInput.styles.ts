import styled from 'styled-components/native';
import { TouchableOpacity } from 'react-native';

import { defaultTextInputIconSize as defaultTextInputContentHeight } from '../TextInput';
import { P } from '../Typography';
import { scale } from '../../utils/layout';

export const VisibilityButton = styled(TouchableOpacity)`
  align-items: center;
  justify-content: center;
  height: ${defaultTextInputContentHeight}px;
  padding-right: ${scale(16)};
`;

export const VisibilityLabel = styled(P).attrs({
  color: 'link',
})`
  text-transform: capitalize;
`;
