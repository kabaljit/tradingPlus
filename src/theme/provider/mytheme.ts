import { colorPalette } from '../palette';
import context from '../context';


const {
  customColors,
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  ...contextRest
} = context;

export const tradeTheme = {
  colorPalette: colorPalette,
  context: {
    colors: customColors,
    ...contextRest,
  },
};
