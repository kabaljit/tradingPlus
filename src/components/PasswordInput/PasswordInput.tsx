import * as React from 'react';
import { TextInput as RNTextInput } from 'react-native';

import TextInput from '../TextInput';

import { VisibilityButton, VisibilityLabel } from './PasswordInput.styles';
import { IPasswordInputProps } from './PasswordInput.models';

export const PasswordInput = React.forwardRef<RNTextInput, IPasswordInputProps>(
  (props, ref) => {
    const { showLabel, hideLabel } = props;

    const [passwordVisible, setPasswordVisibility] = React.useState(false);

    return (
      <TextInput
        ref={ref}
        secureTextEntry={!passwordVisible}
        autoCapitalize="none"
        autoCorrect={false}
        returnKeyType="go"
        accessoryRight={
          <VisibilityButton
            onPress={() => setPasswordVisibility(!passwordVisible)}
          >
            <VisibilityLabel
              testID={passwordVisible ? 'hide-password' : 'show-password'}
              color="link"
            >
              {passwordVisible
                ? hideLabel
                  ? hideLabel
                  : 'Hide'
                : showLabel
                ? showLabel
                : 'Show'}
            </VisibilityLabel>
          </VisibilityButton>
        }
        {...props}
      />
    );
  }
);
