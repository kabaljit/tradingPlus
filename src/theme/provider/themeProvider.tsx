import { createTheming } from "@callstack/react-theme-provider";
import React from "react";

import { tradeTheme } from "./mytheme";
import { ITheme, IThemeProvider } from "./models";

const {
  ThemeProvider: BareThemeProvider,
  withTheme,
  useTheme,
} = createTheming<ITheme>(tradeTheme);

export const ThemeProvider: React.FunctionComponent<IThemeProvider> = ({
  children,
  theme,
}) => <BareThemeProvider theme={theme}>{children}</BareThemeProvider>;

export { useTheme, withTheme };
