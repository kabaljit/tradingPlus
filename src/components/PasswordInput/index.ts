import { withTheme } from '../../theme/provider/themeProvider';

import { PasswordInput } from './PasswordInput';

export default withTheme(PasswordInput);
