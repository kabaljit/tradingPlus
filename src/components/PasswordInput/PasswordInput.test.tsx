import React from 'react';
import renderer from 'react-test-renderer';

import { ThemeProvider } from '../../theme/provider/themeProvider';

import PasswordInput from '.';

describe('Given a PasswordInput', () => {
  describe('when the component is rendered', () => {
    it('should match the snapshot', () => {
      const tree = renderer
        .create(
          <ThemeProvider>
            <PasswordInput />
          </ThemeProvider>
        )
        .toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
