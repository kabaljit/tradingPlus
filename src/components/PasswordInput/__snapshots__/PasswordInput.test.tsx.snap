// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PasswordInput when the component is rendered should match the snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(247, 248, 252, 1)",
      "minHeight": 74.4,
      "position": "relative",
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "position": "relative",
          },
        ]
      }
    >
      <TextInput
        accessoryRight={
          <Styled(Component)
            onPress={[Function]}
          >
            <Styled(withTheme(Paragraph))
              color="link"
              testID="show-password"
            >
              Show
            </Styled(withTheme(Paragraph))>
          </Styled(Component)>
        }
        align="left"
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        rejectResponderTermination={true}
        returnKeyType="go"
        secureTextEntry={true}
        selectionColor="#FF4D54"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "fontSize": 21.25714285714286,
              "fontWeight": "normal",
              "lineHeight": 25.242857142857144,
              "minHeight": 74.4,
              "paddingLeft": 22.58571428571429,
              "textAlign": "left",
            },
          ]
        }
        textAlignVertical="bottom"
        textOnly={true}
        theme={
          Object {
            "colorPalette": Object {
              "green": Array [
                "#0BA866",
              ],
              "grey": Array [
                "#F7F8FC",
                "#EDEDF4",
                "#E1E1EA",
                "#D4D5DD",
                "#BCBCC9",
                "#9699AA",
                "#7C7E8D",
                "#676978",
                "#484A58",
              ],
              "mango": Array [
                "#FFB900",
                "#FFD566",
              ],
              "midnightBlue": Array [
                "#262835",
              ],
              "red": Array [
                "#FFFEF3",
                "#FFE0D7",
                "#FFC2BB",
                "#FFA5A0",
                "#FF8886",
                "#FF6B6C",
                "#FF4D54",
                "#FF333C",
                "#FF1A23 ",
              ],
              "seaGreen": Array [
                "#D5F1F3",
                "#C1EAED",
                "#ACE3E7",
                "#9CDEE1",
                "#82D5DA",
                "#6DCDD3",
                "#59C7CE",
                "#2FB9C2",
                "#238B92",
                "#185D61",
              ],
              "skyBlue": Array [
                "#F7FAFD",
                "#E5F2FC",
                "#C0DEF8",
                "#99CAF4",
                "#75B6EF",
                "#5DA6EC",
                "#4C97E8",
                "#4689DA",
                "#3871C2",
                "#3260B0",
              ],
              "white": Array [
                "#FFFFFF",
              ],
              "yellow": Array [
                "#FFF8E4",
              ],
            },
            "context": Object {
              "colors": Object {
                "actions": Object {
                  "background": Object {
                    "link": Object {
                      "complete": "",
                      "default": "transparent",
                      "disabled": "transparent",
                      "error": "",
                      "focused": "transparent",
                      "loading": "transparent",
                    },
                    "primary": Object {
                      "complete": "",
                      "default": "#FF4D54",
                      "disabled": "#E1E1EA",
                      "error": "",
                      "focused": "",
                      "loading": "#E1E1EA",
                    },
                    "secondary": Object {
                      "complete": "",
                      "default": "transparent",
                      "disabled": "transparent",
                      "error": "",
                      "focused": "transparent",
                      "loading": "transparent",
                    },
                  },
                  "foreground": Object {
                    "link": Object {
                      "complete": "",
                      "default": "#FFFEF3",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                    "primary": Object {
                      "complete": "",
                      "default": "#FFFFFF",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                    "secondary": Object {
                      "complete": "",
                      "default": "#FFFEF3",
                      "disabled": "#262835",
                      "error": "",
                      "focused": "",
                    },
                  },
                },
                "input": Object {
                  "background": Object {
                    "complete": "#F7FAFD",
                    "default": "#F7F8FC",
                    "error": "#FFFEF3",
                    "focused": "#FFFFFF",
                  },
                  "border": Object {
                    "complete": "#4689DA",
                    "default": "#9699AA",
                    "error": "#FFE0D7",
                    "focused": "#FF4D54",
                  },
                  "icon": Object {
                    "complete": "#262835",
                    "default": "#262835",
                    "error": "#262835",
                    "focused": "#262835",
                  },
                  "label": Object {
                    "complete": "#9699AA",
                    "default": "#9699AA",
                    "error": "#FFE0D7",
                    "focused": "#9699AA",
                  },
                  "outerLabel": Object {
                    "complete": "#262835",
                    "default": "#262835",
                    "error": "#262835",
                    "focused": "#262835",
                  },
                },
                "states": Object {
                  "error": Object {
                    "complete": "",
                    "default": "#FFFEF3",
                    "error": "",
                    "focused": "",
                  },
                  "loading": Object {
                    "complete": "",
                    "default": "#FFB900",
                    "error": "",
                    "focused": "",
                  },
                },
                "typography": Object {
                  "blue": "#3260B0",
                  "error": "#FF8886",
                  "inverse": "#D4D5DD",
                  "link": "#3871C2",
                  "mango": "#FFB900",
                  "primary": "#262835",
                  "secondary": "#F7F8FC",
                  "success": "#0BA866",
                  "tertiary": "#7C7E8D",
                  "white": "#FFFFFF",
                },
              },
              "fontSizes": Object {
                "paragraph": Object {
                  "large": 31.5,
                  "normal": 27.5,
                  "small": 23.5,
                },
                "title": Object {
                  "large": 63,
                  "normal": 47.5,
                  "small": 35.5,
                },
              },
              "fontWeights": Object {
                "paragraph": Object {
                  "bold": "700",
                  "medium": "600",
                  "regular": "400",
                },
                "title": Object {
                  "bold": "700",
                  "medium": "600",
                  "regular": "400",
                },
              },
              "sizes": Object {
                "buttonHeight": 108.5,
                "formElementHeight": 91,
                "formElementIconHeight": 67,
                "hairline": 0.5,
                "listItemHeight": 108.5,
              },
              "spacing": Object {
                "element": 16,
                "screenEdge": 31.5,
              },
            },
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
    <View
      testID="accessory-right"
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 58.78928571428571,
            "justifyContent": "center",
            "opacity": 1,
            "paddingRight": 21.25714285714286,
          }
        }
      >
        <Text
          selectable={false}
          style={
            Object {
              "color": "#3871C2",
              "fontFamily": "System",
              "fontSize": 21.25714285714286,
              "fontWeight": "400",
              "lineHeight": 29.76,
              "textAlign": "left",
            }
          }
        >
          Show
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(150, 153, 170, 1)",
        "bottom": 0,
        "height": 2.6571428571428575,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "width": "100%",
      }
    }
  />
</View>
`;
